include/bluetooth/mesh/gen_prop_cli.h

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @defgroup bt_mesh_prop_cli Generic Property Client model
+ * @{
+ * @brief API for the Generic Property Client model.
+ */
+#ifndef BT_MESH_GEN_PROP_CLI_H__
+#define BT_MESH_GEN_PROP_CLI_H__
+
+#include <bluetooth/mesh/gen_prop.h>
+#include <bluetooth/mesh/model_types.h>
+
+struct bt_mesh_prop_cli;
+
+/** @def BT_MESH_PROP_CLI_INIT
+ *
+ * @brief Initialization parameters for the @ref bt_mesh_prop_cli.
+ *
+ * @param[in] _prop_list_handler Optional status message handler.
+ * @sa bt_mesh_prop_cli::prop_list
+ * @param[in] _prop_status_handler Optional status message handler.
+ * @sa bt_mesh_prop_cli::prop_status
+ */
+#define BT_MESH_PROP_CLI_INIT(_prop_list_handler, _prop_status_handler)        \
+	{                                                                      \
+		.pub = { .msg = NET_BUF_SIMPLE(BT_MESH_MODEL_BUF_LEN(          \
+				 BT_MESH_PROP_OP_ADMIN_PROP_SET,               \
+				 BT_MESH_PROP_MSG_MAXLEN_ADMIN_PROP_SET)) },   \
+		.prop_list = _prop_list_handler,                               \
+		.prop_status = _prop_status_handler,                           \
+	}
+
+/** @def BT_MESH_MODEL_PROP_CLI
+ *
+ * @brief Generic Property Client model composition data entry.
+ *
+ * @param[in] _cli Pointer to a @ref bt_mesh_prop_cli instance.
+ */
+#define BT_MESH_MODEL_PROP_CLI(_cli)                                           \
+	BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_GEN_PROP_CLI, _bt_mesh_prop_cli_op,  \
+			 &(_cli)->pub,                                         \
+			 BT_MESH_MODEL_USER_DATA(struct bt_mesh_prop_cli,      \
+						 _cli),                        \
+			 &_bt_mesh_prop_cli_cb)
+
+/** List of property IDs. */
+struct bt_mesh_prop_list {
+	u8_t count; /**< Number of IDs in the list. */
```

Comment:
Use size_t?

Reply:
I don't think that'd convey the right intention. `size_t` is associated with `sizeof`, and normally refers to memory size. In this case, it's the number of elements in the array, so I'd like to keep it as is.